### PR TITLE
Enable close of main window

### PR DESCRIPTION
**Pull request by dalonsoa**
Closing the main window is a menu entry that does work in MacOS - because it adds its own entries - but not in Windows or Linux. This PR adds an explicit callback to the `Exit` entry to close the window instead of relying on the `ID`, which only works for buttons. See [this comment](https://stackoverflow.com/questions/28103710/wxpython-fails-to-quit/28111788#28111788).

As a side effect, given that the menu entry `Exit` had to interact with the main Window, I move it to the main class. It is not ideal as it makes it a special case, but the alternative is passing the main window instance to the menu creator, which will break the backwards compatibility (which is fine, but better doing it just once in a while).

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -23,7 +23,7 @@
 author = "Research Computing Service, Imperial College London"
 
 # The full version, including alpha/beta/rc tags
-release = "0.2.2"
+release = "0.2.3"
```
Comment (alexdewar) [inline docs/conf.py:26]: Is this an unintentional change? Isn't `guikit` *already* on v0.2.3?

File: guikit/core.py [hunk 2]
```diff
@@ -95,11 +95,28 @@ def populate_window(self):
         self._make_menubar()
         self.SetInitialSize(wx.Size(self.size))
 
+    def on_quit(self, evt):
+        """Event to close the main window form the menu."""
```
Comment (alexdewar) [inline guikit/core.py:99]:
```suggestion
        """Event to close the main window from the menu."""
```

File: guikit/core.py [hunk 3]
```diff
@@ -95,11 +95,28 @@ def populate_window(self):
...
+        self.Close()
+
+    def populate_built_in_menu(self):
+        """Populate the menu with some built-in capabilities."""
+        return [
+            MenuTool(
+                menu="File",
+                id=wx.ID_EXIT,
```
Comment (alexdewar) [inline guikit/core.py:107]: Do we want/need to set ``id`` here? The link you provided seems to suggest that ``wx.ID_EXIT`` is for dialogs only in any case: https://stackoverflow.com/questions/28103710/wxpython-fails-to-quit/28111788#28111788

File: guikit/core.py [hunk 4]
```diff
@@ -95,11 +95,28 @@ def populate_window(self):
...
+            ),
+        ]
+
     def _make_menubar(self) -> None:
         """Create the menu bar from the entries provided by the widgets."""
         # Collecting the menu entries
         entries = itertools.chain.from_iterable(
-            [view().menu_entries() for view in KNOWN_PLUGINS]
+            [self.populate_built_in_menu()]
+            + [view().menu_entries() for view in KNOWN_PLUGINS]
```
Comment (alexdewar) [inline guikit/core.py:119]: Does this mean that the "exit" button will always be at the top of the "File" menu? Isn't it usually at the bottom?

**Review by alexdewar — COMMENTED** [review]
I've got some queries, but otherwise looking good. Thanks for looking at this!

I'm still not entirely clear how things work on macOS though. Am I right in thinking that there would be an exit/quit menu item independent of whether we add one explicitly with `guikit`? Do you end up with two of them if there's one there already and we add another?

**Comment by dalonsoa** [comment]
I've done some changes and updated `develop` from `main`, which is the reason it appeared to have changes in the version. I've also made sure there's only one `Quit` menu entry in MacOS and that this one is last, but as I mention, there's an issue open about the order of entries.

**Review by alexdewar — APPROVED** [review]
LGTM! :smile:

The order of entries is an interesting problem, but you're right it's probably one for another day!

**Comment by alexdewar** [comment]
PS -- Maybe we should make another release after this is merged as it is a bugfix? It would be handy for iPANACEA.

**Comment by dalonsoa** [comment]
Yep, that's the plan

**Pull request merged**